proposals/extend_pod_security_admission.md

```diff
@@ -0,0 +1,229 @@
+# Extending Pod Security Admission
+
+**Author**: Shuting Zhao (shuting@nirmata.com)
+**Date**: Dec 15th, 2021
+**Update**: June 15th, 2022
+**Abstract**: Using Kyverno to extend Pod Security Admission for Kubernetes.
+
+## Contents
+
+  * [Contents](#contents)
+  * [Introduction](#introduction)
+  * [The Problem](#the-problem)
+  * [Proposed Solution](#proposed-solution)
+    + [Extend PSA via a new rule](#extend-psa-via-a-new-rule)
+    + [Exceptions for container security context](#exceptions-for-container-security-context)
+  * [Alternate Solutions Considered](#alternate-solutions-considered)
+    + [Extend Via Annotations](#extend-via-annotations)
+      - [Step 1. Enable additional policies for a namespace](#step-1-enable-additional-policies-for-a-namespace)
+      - [Step 2. Fine-grained Policy Control](#step-2-fine-grained-policy-control)
+  * [Next Steps](#next-steps)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+
+## Introduction
+
+[Pod Security admission](https://kubernetes.io/docs/concepts/security/pod-security-admission/) (PSA) is a built-in solution that applies different isolation levels of [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) for Pods. With the release of Kubernetes v1.23, PSA has entered beta and is enabled by default for 1.23+ clusters. 
+
+For more information on using PSA, see: [how PSA is used in a cluster](https://hackmd.io/C0prhSfdTbSv9d1ag7Ft9g).
+
+This document proposes a solution to extend PSA for finer-grained and flexible policy control.
+
+## The Problem
+
+Once PSA is enabled for namespaces, a configured level of Privileged, Baseline, or Restricted applies to all pods and workloads within the namespace. The level is configured as a label on the namespace. There is no option to select specific pods or control, for granular policies. 
+
+Users can choose to configure controls for select pods, but there is no validation or enforcement beyond meeting the requirements for the specified level. This means that if a namespace is baseline, any pod in that namespace can run with root user privileges. 
+
+Here is an example of this request: [Grant specific permissions for specific services](https://github.com/kubernetes/pod-security-admission/issues/1).
+
+A solution would be to use microsegmentation, and isolate privileged pods to their own namespace. However, this introduces other complexities such as managing networking across pods across namespaces, managing permissions, etc.
+
+There are a few other considerations with PSA:
+
+* For users, violations are visible only for `warn` and `enforce` mode. With `audit` there's no trace that users can track unless the audit trace is enabled via the API-server.
```

Comment:
And even in `audit` mode, users who create the resources can't track them. Only someone that has access to the audit log (a control plane, protected feature) can.

Reply:
Rephrased.